### PR TITLE
Refine cell filtering UI (SCP-5297)

### PR DESCRIPTION
**Pull request by eweitz**
This smoothens a few aspects of the prototype cell filtering UI:

* Fix background color in cell filtering panel
* Enable select / deselect by clicking filter label (not just checkbox)
* Only show loading indicators for cell filtering once button is clicked, like DE loading UX -- this minimizes unnecessary visual distraction, and fixes size issues with the prior loading animation
* Omit user annotations from cell facets (for now)
* Refined user-facing text

It also makes a handful of internal improvements for maintainability, like SCP-consistent function declarations, clearer and less abbreviated names, and more IDE-friendly docstrings.

Here's how it looks.

### Before

https://github.com/broadinstitute/single_cell_portal_core/assets/1334561/901e34d2-1bff-482f-ae1f-63caa9b45e9d

### After

https://github.com/broadinstitute/single_cell_portal_core/assets/1334561/7c8d3c84-8257-4800-9d73-0eb48e445b58

Known issues that are not in scope here include:
* Groups can unexpectedly change color upon filtering cluster scatter plot
* Unexpected filtering UI upon changing to non-default annotation
* Not all applicable annotations available in facet dropdown menus

### Test
* Enable filtering via feature flag
* Go to "Human milk - DE pilot" / "Human lactation" study
* Click "Cell filtering" button immediately upon seeing it
* Confirm you can click it, and upon click a loading UI displays in the cell filtering panel
* Wait for filters to load (this is considerably slower in development)
* Confirm panel back button has same color as rest of filtering panel, like DE panel
* Click a filter label
* Confirm that selects the associated checkbox and filters the cluster scatter plot

This relates to SCP-5297.

### CODE REVIEW

**Comment by eweitz** [comment]
A [build failure](https://github.com/broadinstitute/single_cell_portal_core/actions/runs/6394820302/job/17357085467#step:5:4024) was a false positive, as only Rails tests failed but no Rails code changed.

```
Total elapsed time: 20 minutes, 0 seconds

### FAILURES and ERRORS ###
*** Yarn test failures ***

*** Rails test failures ***

  StudiesControllerTest#test_should_get_study_file_info_hash:
  NameError: "my_flag" is not a valid feature flag name

  ClustersControllerTest#test_enforces_view_permissions:
  Mongoid::Errors::Validations: 

  StudyValidationTest#test_should_allow_files_with_spaces_in_names [/home/app/webapp/test/integration/external/study_validation_test.rb:281]:
  Expression upload failed: 302.

  StudyValidationTest#test_should_fail_all_ingest_pipeline_parse_jobs [/home/app/webapp/test/integration/external/study_validation_test.rb:54]:
  Did not redirect to upload successfully.

  StudyValidationTest#test_should_prevent_changing_firecloud_attributes [/home/app/webapp/test/integration/external/study_validation_test.rb:165]:
  Expected at least 1 element matching "li#study_error_firecloud_project", found 0.

  StudyValidationTest#test_should_disable_downloads_for_reviewers:
  Mongoid::Errors::Validations: 

  StudyValidationTest#test_should_validate_unique_cells_for_expression_matrices [/home/app/webapp/test/integration/external/study_validation_test.rb:314]:
  Expression matrix upload failed: 302.

  StudyValidationTest#test_should_enforce_access_restrictions_on_editing_studies [/home/app/webapp/test/integration/external/study_validation_test.rb:346]:
  Expected response to be a <2XX: success>, but was a <302: Found> redirect to <http://www.example.com/single_cell/users/sign_in>

  StudyValidationTest#test_should_sanitize_filenames_with_special_characters [/home/app/webapp/test/integration/external/study_validation_test.rb:299]:
  README failed: 302.

  StudyValidationTest#test_should_delete_data_from_bigquery:
  NoMethodError: undefined method `[]' for nil:NilClass

  StudyFilesControllerTest#test_should_parse_study_file [/home/app/webapp/test/api/study_files_controller_test.rb:126]:
  Expected response to be a <204: No Content>, but was a <401: Unauthorized>

  StudyFilesControllerTest#test_can_update_cluster_name [/home/app/webapp/test/api/study_files_controller_test.rb:99]:
  Expected: "updated name"

  StudyFilesControllerTest#test_should_create_then_update_then_delete_study_file [/home/app/webapp/test/api/study_files_controller_test.rb:74]:
  Expected response to be a <2XX: success>, but was a <401: Unauthorized>

  StudyFilesControllerTest#test_should_create_and_update_AnnData_file [/home/app/webapp/test/api/study_files_controller_test.rb:257]:
  Expected response to be a <2XX: success>, but was a <401: Unauthorized>

  SearchControllerTest#before_all:
  Mongoid::Errors::Validations: 

  UploadCleanupJobTest#test_should_only_run_cleanup_job_3_times_on_error:
  NoMethodError: undefined method `handler' for nil:NilClass

  StudyFileBundleTest#before_all:
  Mongoid::Errors::Validations: 

  DifferentialExpressionServiceTest#before_all:
  Mongoid::Errors::Validations: 

  StudyTest#before_all:
  Mongoid::Errors::Validations: 

  StudyFileTest#before_all:
  Mongoid::Errors::Validations: 
Exiting with code: 1
```

**Review by jlchang — APPROVED** [review]
Functional review - looks great! 

I really like the user-facing text changes, much more succinct.

**Review by bistline — APPROVED** [review]
Looks good - small question re: help text.

File: app/javascript/components/explore/FacetFilteringModal.jsx [hunk 1]
```diff
@@ -9,24 +9,23 @@ export default function FacetFilteringModal() {
...
-        You can visualize the expression of particular cells in a subset of the dataset through the facet filters.
+      <p>
+        Cell filtering lets you easily plot cells that match criteria <i>across annotations</i>.
       </p>
       <p>
-        For example - You might be interested in the diseased cells of a dataset, but only in the males of the dataset. You can
-        choose any annotation to color your plot by and then chose facets Disease and Sex then filter for only diseased and male
-        cells utilizing the check boxes to filter.
+        For example, cell filtering can let you quickly subset a plot to show only cells that are diseased <i>and</i> from males.
+        You would click to update the filters so only "Yes" is checked under "Disease" and "Male" is checked under "Sex".
```
[the remark this answers is not in the document]
Comment (eweitz) [reply app/javascript/components/explore/FacetFilteringModal.jsx]: Good suggestion.  I've updated it with a specific disease -- lymphoma, as in our [metadata explainer](https://github.com/broadinstitute/single_cell_portal_core/blob/development/app/assets/images/metadata-convention-explainer.jpg).

**Pull request merged**